Assets/MixedRealityToolkit/InputSystem/Sources/GestureInputSource.cs

```diff
@@ -0,0 +1,191 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Internal.Definitions.InputSystem;
+using Microsoft.MixedReality.Toolkit.Internal.Definitions.Utilities;
+using UnityEngine.XR.WSA.Input;
+
+namespace Microsoft.MixedReality.Toolkit.InputSystem.Sources
+{
+    /// <summary>
+    /// Input source supporting input from gesture recognizer.
+    /// </summary>
+    public class GestureInputSource : BaseGenericInputSource
+    {
+        /// <summary>
+        /// Pointer Action for Gesture Tap or "Click"
+        /// </summary>
+        public static MixedRealityInputAction PointerAction { get; set; } = MixedRealityInputAction.None;
+
+        /// <summary>
+        /// Hold action to use when events are raised.
+        /// </summary>
+        public static MixedRealityInputAction HoldAction { get; set; } = MixedRealityInputAction.None;
+
+        /// <summary>
+        /// Manipulation action to use when events are raised.
+        /// </summary>
+        public static MixedRealityInputAction ManipulationAction { get; set; } = MixedRealityInputAction.None;
+
+        /// <summary>
+        /// Navigation action to use when events are raised.
+        /// </summary>
+        public static MixedRealityInputAction NavigationAction { get; set; } = MixedRealityInputAction.None;
+
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="useRailsNavigation">Should the gesture input source use rails navigation?</param>
+        public GestureInputSource(bool useRailsNavigation) : base("Gesture Input Source")
+        {
+            gestureRecognizer = new GestureRecognizer();
+
+            gestureRecognizer.Tapped += GestureRecognizer_Tapped;
+
+            gestureRecognizer.HoldStarted += GestureRecognizer_HoldStarted;
+            gestureRecognizer.HoldCompleted += GestureRecognizer_HoldCompleted;
+            gestureRecognizer.HoldCanceled += GestureRecognizer_HoldCanceled;
+
+            gestureRecognizer.ManipulationStarted += GestureRecognizer_ManipulationStarted;
+            gestureRecognizer.ManipulationUpdated += GestureRecognizer_ManipulationUpdated;
+            gestureRecognizer.ManipulationCompleted += GestureRecognizer_ManipulationCompleted;
+            gestureRecognizer.ManipulationCanceled += GestureRecognizer_ManipulationCanceled;
+
+            gestureRecognizer.SetRecognizableGestures(GestureSettings.Tap | GestureSettings.ManipulationTranslate | GestureSettings.Hold);
```

Comment:
seems as if we are forcing apps to register for all of the gestures... they should be allowed to limit their selection (saves the recognizer unneeded work)

Reply:
I agree, we should make this configurable as well.